crates/ra_hir_ty/src/tests.rs

```diff
@@ -344,3 +345,29 @@ fn typing_whitespace_inside_a_function_should_not_invalidate_types() {
         assert!(!format!("{:?}", events).contains("infer"), "{:#?}", events)
     }
 }
+
+// Infer with some common definitions and impls.
+fn check_infer(ra_fixture: &str, expect: Expect) {
+    let defs = r#"
+        #[lang = "sized"]
+        pub trait Sized {}
+        #[lang = "unsize"]
+        pub trait Unsize<T: ?Sized> {}
+        #[lang = "coerce_unsized"]
+        pub trait CoerceUnsized<T> {}
+
+        impl<'a, 'b: 'a, T: ?Sized + Unsize<U>, U: ?Sized> CoerceUnsized<&'a U> for &'b T {}
+        impl<T: ?Sized + Unsize<U>, U: ?Sized> CoerceUnsized<*mut U> for *mut T {}
```

Comment:
I'm not a fan of adding these definitions globally -- I don't like adding any definitions implicitly in all tests, and these particular ones are only useful in some of the coercion tests anyway. (Also, this seems like it would be additionally confusing in multi-file fixtures...)

Reply:
I thought we were moving towards a "small `std`" as a fixture in other tests, and this existing code anyway. But I'll try to remove it.

Reply:
Done,